### ISSUE TITLE
Preview feature feedback: Find a relation's count (`selectRelationCount `)

### ISSUE THREAD

**Issue by matthewmueller** [prisma/prisma#6312]
Please share your feedback about the `selectRelationCount` functionality released in [v2.20.0](https://github.com/prisma/prisma/releases/tag/2.20.0) in this issue.

- If you encounter a bug, please [open a bug report](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=bug_report.md&title=) in this repo. 
- If the feature is working well for you, please share this in a comment below or leave a 👍 on this issue.

If you have any questions, don't hesitate to ask them in the [`#prisma-client`](https://prisma.slack.com/archives/CCWDULGUW) channel in the [Prisma Slack](https://slack.prisma.io).

**Comment by janpio** [comment]
There is one bug at https://github.com/prisma/prisma/issues/6543 that was reported multiple times now.

**Comment by 41y08h** [comment]
> There is one bug at #6543 that was reported multiple times now.

Nested relation count is the feature that is not working and produces this error.
Hoping to see prisma resolve it fast.

Thanks.

**Comment by msahajwani** [comment]
Hi,

I have this query:

```
const user = await prisma.user.findFirst({
            where: { username: username },
            include: {
                posts: {
                    include: {
                        likedBy:true,
                    } 
                },
            }
        })
```

I’m getting a user with a specific user name, and I’m also getting the posts for the user. And for each post, I’m getting likedBy (which is a relation).

This works perfectly.

However, instead of getting the array of likedBy, if I want to get the count of liked by, I do the following:

```
const user = await prisma.user.findFirst({
            where: { username: username },
            include: {
                posts: {
                    include: {
                        _count: {
                            select: {
                                likedBy:true
                            }
                        }
                    } 
                },
            }
        })
```

This query fails with error: PANIC in query-engine.

Looks like: Currently the _count parameter can only be used on a top level include or select and not nested levels.

Can this be supported, please.

**Comment by SamuelMS** [comment]
Opened #7688 to track this issue, but the `_count` field doesn't seem to be supported by `Prisma.validator` and `Prisma.XYZGetPayload` for partial / related types.

```javascript
// works fine
const user = Prisma.validator<Prisma.UserArgs>()({
  select: { id: true, name: true },
});
export type User = Prisma.UserGetPayload<typeof user>;

// TS2322: Type 'boolean' is not assignable to type 'never'.
const userWithCount = Prisma.validator<Prisma.UserArgs>()({
  select: { id: true, name: true, _count: { select: { posts: true } },  },
});
```

**Comment by zfanta** [comment]
Hi,

I hope the fields of `SomeCountOutputTypeSelect` support not only boolean type.

I have this schema.
```
model Post {
  id      String   @id @db.VarChar(32)
  body    String   @db.Text
  reports Report[]
}

model Report {
  id       String  @id @db.VarChar(32)
  postId   String? @db.VarChar(32)
  post     Post?   @relation(fields: [postId], references: [id], onDelete: Cascade)
  reason   String? @db.Text
  validity Boolean @default(true)

  @@index([postId])
}
```

And I have to count the valid reports of post. So I use this code that manually calls count function for every post.
```
const post = await prisma.post.findUnique({
  where: { id: 'some id' },
  include: {
    _count: {
      select: {
        reports: true
      }
    }
  }
})
                                                 
post._count.reports = await prisma.report.count({
  where: {
    postId: 'some id',
    validity: true
  }
})
```

If the fields of `SomeCountOutputTypeSelect` are `boolean|SomeWhereInput`, I can use this code.
```
const post = await prisma.post.findMany({
  where: { id: 'some id' },
  include: {
    _count: {
      select: {
        reports: {
          validity: true
        }
      }
    }
  }
})
```

**Comment by andrelandgraf** [comment]
Is this preview feature also supported with mongoDB?

**Comment by 2color** [comment]
@andrelandgraf You cannot currently use `selectRelationCount` with the MongoDB connector.

**Comment by 2color** [comment]
Closing this now that this has been released to General Availability in the [3.0.1 release](https://github.com/prisma/prisma/releases/tag/3.0.1)
